frontend/src/components/worksheets/Worksheet.js

```diff
@@ -718,8 +718,10 @@ class Worksheet extends React.Component {
                         // If the number of bundles decreases, then focus should be on the same bundle as before
                         // unless that bundle doesn't exist anymore, in which case we select the closest bundle that does exist,
                         // where closest means 'next' by default or 'last' if there is no next bundle.
-                        var focus = this.getFocusAfterBundleRemoved();
+                        var focus = this.getFocusAfterBundleRemoved(items);
                         this.setFocus(focus[0], focus[1]);
+                    } else {
+                        this.setFocus(this.state.focusIndex + 1, this.state.subFocusIndex);
```

Comment:
Overall looks good, can you add a comment here explaining when this else condition happens and why the setFocus indices? (like the comments from 718-720) The code is getting more complex so would be good to have comments to follow

Reply:
Thanks Yipeng!